packages/teleport/src/cluster/components/Users/UserDialog/UserDialog.tsx

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2020 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import copyToClipboard from 'design/utils/copyToClipboard';
+import selectElementContent from 'design/utils/selectElementContent';
+import { ButtonPrimary, ButtonSecondary, Alert, Text, Flex } from 'design';
+import Dialog, {
+  DialogHeader,
+  DialogTitle,
+  DialogContent,
+  DialogFooter,
+} from 'design/Dialog';
+import Validation from 'shared/components/Validation';
+import FieldInput from 'shared/components/FieldInput';
+import FieldSelect from 'shared/components/FieldSelect';
+import { Option } from 'shared/components/Select';
+import { requiredField } from 'shared/components/Validation/rules';
+import { InviteToken } from 'teleport/services/user';
+import useUserDialog from './useUserDialog';
+import { DialogState } from '../useUsers';
+
+export default function StateWrapper(props: Omit<Props, 'state'>) {
```

Comment:
We can call it `Container` - some term used to indicate that given component manages its own state.

Reply:
revised